netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -108,17 +107,17 @@
       long maxConnectionIdleInNanos,
       long maxConnectionAgeInNanos, long maxConnectionAgeGraceInNanos,
       boolean permitKeepAliveWithoutCalls, long permitKeepAliveTimeInNanos,
-      InternalChannelz channelz) {
+      InternalChannelz channelz, boolean usingSharedBossGroup, boolean usingSharedWorkerGroup) {
```

Comment:
Having these bools passed in this deep can be sort of annoying. You may consider passing in ObjectPools instead. That way this code wouldn't need any conditionals and some of the handling elsewhere looks like it would improve.

Reply:
done. it looks much simpler